### ISSUE TITLE
Create "Current Node" block

### ISSUE THREAD

**Issue by quicksketch** [backdrop/backdrop-issues#5774]
## Description of the need

We have an existing block `NodeBlock` that allows pointing at any arbitrary node as a block by its Node ID. I have a situation where the best way to use layouts may be to render the same node multiple times on the same page, each with different display modes. For example I have a banner area at the top of the page, as well as the main content area that is a wide column. I want to display a "banner" display mode in the banner area in addition to the "full page' display mode used in the main content block.

## Proposed solution

Introduce `NodeCurrentBlock` class that can render the node. This block would have the required context of `node` so it would only show up on `node/%` path layouts (or other layouts with a Node context).

**Comment by indigoxela** [comment]
> I have a situation where the best way to use layouts may be to render the same node multiple times on the same page...

That's interesting. :thinking: I wonder, why field blocks don't do the trick in that case.

Something like:

- A layout with `node/%` and visibility condition on your node type
- Place body, banner, ... as field blocks in their layout regions

Maybe your display modes do something more fancy than a single field block can do? Possibly views could cover that?

**Comment by quicksketch** [comment]
Here's the design for reference. The "Top Region" I was planning to make a display mode for all the fields present:

* Title
* Related Functions
* Download Link
* Primary Image

The breadcrumb and Previous/Next links would be either a single block, or two separately placed ones with some positioning CSS.

![image](https://user-images.githubusercontent.com/100206/189830791-bf16300d-3f5b-4708-a691-52e57d2f8871.png)

In my case, I have 4 different node types that have different fields, but similar designs for this top region. I'd like to use one layout for all 4 node types, rather than needing to position the fields individually in separate layouts (or putting fields that have no content in one layout). Using a display mode hopefully reduces the chance of breaking the page layout and having 6-8 fields across the 4 types all pulled out and displayed individually.

As you point out, this could also be done with a View that runs a query to select _only_ the current node on the page via a contextual filter. Then within the View I could either select individual fields or just use a display mode to render the node again.

I guess the feature isn't critical since it can already be achieved by routing the Views, but it's another step and one that shouldn't be necessary.

**Comment by smaiorana** [comment]
Hello, we're running into this same problem. We have a couple of Drupal 7 sites that use Panels. Several of those Panels display entire nodes based on context. We're in the process of converting our sites to Backdrop and trying to replicate the same functionality using Layouts, but the ability to display an entire node based on context doesn't exist.

We're trying to replicate the functionality using Views, but we can't figure out how to pass context from a Layout into a View. I looked through core code, and it appears the context handling option for View blocks is marked as @TODO.

I know we can extract context from the URL based on position, but this is not ideal. How did you replicate this same functionality with the existing Layout module in conjunction with Views?

**Comment by quicksketch** [comment]
> I know we can extract context from the URL based on position, but this is not ideal. How did you replicate this same functionality with the existing Layout module in conjunction with Views?

Yeah, I think loading by position is the only way currently to render a display mode. However it seems like this request might be too niche for core(?) I ended up putting this functionality into a custom module, which might make sense to move to a contrib module.

**Comment by smaiorana** [comment]
> Yeah, I think loading by position with a view is the only way currently to render a display mode. However it seems like this request might be too niche for core(?) I ended up putting this functionality into a custom module, which might make sense to move to a contrib module.

Thanks. We can get around the "Current Node" situation with some custom code if need be. Though a contrib module would certainly be appreciated.

Also, not to hijack the discussion, but are there plans to introduce context handling to View blocks? We have several other Panels that pass context into Views for other reasons. This would also make the "Views" workaround for this issue easier to manage. I can open a separate issue if that's preferable.

**Comment by quicksketch** [comment]
Let's please make a separate issue. I'd love to see that also, since I don't like loading by path position either. And it would allow (in theory) passing arguments from Layout relationships.

**Comment by smaiorana** [comment]
> Let's please make a separate issue. I'd love to see that also, since I don't like loading by path position either. And it would allow (in theory) passing arguments from Layout relationships.

Sure. Issue #5995 opened.

**Comment by hosef** [comment]
@quicksketch Did you end up building this for the site you were working on?

**Comment by yorkshire-pudding** [comment]
There is now a contrib module that allows you to add a display mode of the current node (or other entity that has a layout context):

- https://backdropcms.org/project/current_content_display_block
- https://github.com/backdrop-contrib/current_content_display_block

**Comment by herbdool** [comment]
I believe there is a place to provide a "Use current content" option as part of the "Existing content" block. This is a small tweak and isn't meant to supplant the more complete `current_content_display_block` module.

If checked it hides the "Content title or ID" field:

<img width="796" height="635" alt="Image" src="https://github.com/user-attachments/assets/5b07e56b-6277-4a76-914b-946140bde07c" />

It displays an updated message, for example: "Displays *current content* with *Teaser* display mode."

<img width="839" height="109" alt="Image" src="https://github.com/user-attachments/assets/2765c6bb-eaa2-4902-bfc4-0473ed282273" />

Since I was making changes, I also improved the description if using a selected node, by adding the title and the display mode:

<img width="837" height="108" alt="Image" src="https://github.com/user-attachments/assets/0df15b62-2598-42cb-a4f0-de86b7514eb3" />
